keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1178 @@
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `RecursiveReadOnlyMounts`
+  - Components depending on the feature gate: kube-apiserver,kubelet
+<!--
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+-->
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes, by unsetting `RecursiveReadOnly=Enabled`.
+
+Components can be downgraded too, but it should be noted that `VolumeMountStatus`
+may still see an inconsistent state when kubelet was downgraded.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Works
```

Comment:
Could you expand on this at all?  Is there any potential impact or noticeable behavior?

Reply:
Reworded

> 
> ###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
> Yes, by unsetting `RecursiveReadOnly=Enabled`.
> 
> Components can be downgraded too, but it should be noted that `VolumeMountStatus`
> may still see an inconsistent state when kubelet was downgraded.
> The pod manifest has to be recreated to get a consistent state in this case.
> 
> ###### What happens if we reenable the feature if it was previously rolled back?
> 
> Works.
> Just same as a fresh roll-out, as long as the user has recreated the pod manifests.
> (See "Can the feature be disabled once ..." section above)